x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_duration/register_transaction_duration_rule_type.ts

```diff
@@ -152,6 +154,9 @@ export function registerTransactionDurationRuleType({
                 ...termQuery(TRANSACTION_TYPE, ruleParams.transactionType, {
                   queryEmptyString: false,
                 }),
+                ...termQuery(TRANSACTION_NAME, ruleParams.transactionName, {
+                  queryEmptyString: false,
```

Comment:
hm, that's because I pass an empty string when selecting `All` option 
 instead of an undefined. 
https://github.com/elastic/kibana/pull/154241/files#diff-a7f8b5557da500e95cb8518734a51b96f3c160506b5054f02ba3698a447e9bedR118 

I will update it the component.

Reply:
https://github.com/elastic/kibana/pull/154241/commits/fce4ef8168429645a01434e19b0feaefba1a4f02